### DISCUSSION TITLE
embedded server - websocket and/or XMLHttpRequests possible?

### DISCUSSION THREAD

**Discussion by frenchfaso** [copy/v86#560] [General]
First things first:
**v86 is awesome!**

I played with the lua.html example and that got me thinking...
would some sort of javascript <--> v86 standard network communication be possible?
Something like websockets or XMLHttpRequest/fetch.
That would enable a "client - embedded server" workflow, one could for instance run a nodejs/pyrhon rest-api service in v86 and consume it from JavaScript all from the same static html page.
It would make it super easy to exploit the huge linux apps/utilities ecosystem without the need to wrestle with WASM porting/compilation..

**Comment by copy** [comment]
It's possible, but you we'll need at tcp/ip stack (for example https://github.com/smoltcp-rs/smoltcp). Then fetch could be implemented by a regular http client implementation calling into smoltcp, which in turn sends ethernet packets to the VM.

Some ideas have been discussed here: https://github.com/copy/v86/issues/198

**Reply by frenchfaso** [reply]
Hy @copy 
thanks for the reply!
This is a bit over my head :sweat_smile: should smoltcp (compiled as a WASM module) sit between javascript (window context) and v86?
How could smoltcp communicate with the virtual nic inside v86?
